RadixWallet/Features/AssetTransferFeature/Components/TransferAccountList/ReceivingAccount/Asset/FungibleResourceAsset+Reducer.swift

```diff
@@ -77,6 +110,61 @@ public struct FungibleResourceAsset: Sendable, FeatureReducer {
 
 		case .removeTapped:
 			return .send(.delegate(.removed))
+
+		case let .alertAction(action):
+			state.alert = nil
+			switch action {
+			case let .presented(.chooseXRDAmountAlert(.deductFee(amount))),
+			     let .presented(.chooseXRDAmountAlert(.sendAll(amount))),
+			     let .presented(.needsToPayFeeFromOtherAccount(.confirm(amount))):
+				state.transferAmount = amount
+				state.transferAmountStr = amount.formattedPlain(useGroupingSeparator: false)
+				return .send(.delegate(.amountChanged))
+
+			case .presented(.needsToPayFeeFromOtherAccount(.cancel)):
+				return .none
+
+			case .dismiss:
+				return .none
+			}
 		}
 	}
 }
+
+extension AlertState where Action == FungibleResourceAsset.ViewAction.AlertAction {
```

Comment:
Just to be safe, can you use explicit types instead of .inits, those can sometimes slow down typechecking